src/ContainerCheckpointModal.jsx

```diff
@@ -21,20 +27,43 @@ class ContainerCheckpointModal extends React.Component {
             this.setState({ [event.target.name]: event.target.checked });
     }
 
+    handleCheckpointContainer(args) {
+        const Dialogs = this.context;
+        const container = this.props.containerWillCheckpoint;
+        this.setState({ inProgress: true });
+        client.postContainer(container.isSystem, "checkpoint", container.Id, args)
+                .catch(ex => {
+                    const error = cockpit.format(_("Failed to checkpoint container $0"), container.Names);
+                    this.props.onAddNotification({ type: 'danger', error, errorDetail: ex.message });
+                })
+                .finally(() => {
+                    this.setState({ inProgress: false });
+                    Dialogs.close();
```

Comment:
The `inProgress: false` should be in the error path (catch) when the dialog stays open, to show the inline error. Setting it right before closing the dialog (in the success case) is a no-op, isn't it?

Reply:
Yes, `this.setState({ inProgress: false });` is a no-op and its presence is probably too confusing. The dialog always closes, even when there was an error.  The error message appears in a global notification, not an inline alert.

We have two basic patterns for what happens after the user clicks "Apply" in a dialog (see below), and this dialog here is a mix of them. I learned about this from @garrett just recently, so I might be wrong, as always. :-) Let me try to write this down, I'll copy into some more offical place laterish.

1) The action triggered by "Apply" is carried out while the dialog is open.  All inputs/buttons/checkboxes etc in the dialog are disabled, except` for a "Cancel" or "Stop" button if the action can be cancelled.  A spinner appears in the "Apply" button, and maybe some progress messages appear in the footer. When the action finishes successfully, the dialog is closed.  When there is an error, the error appears in the dialog at the top in a inline alert (and the dialog stays open, of course).

2) The dialog closes as soon as "Apply" is clicked (and input validation has succeeded) and the action runs in the background. There is some immediate indication in the page that the action has started, and some obvious effect when the action has finished successfully. There is no potential for problems when the user interacts with the page while the action is still running. There is no potential for problems when multiple actions run concurrently. If there is an error, a global notification appears. In the case of an error, it is trivial for the user to try again (without having to fill out a big dialog again from scratch).

Pattern 2) is preferred and should be used when all the conditions are fulfilled.

Here, checkpointing a container does not leave a immediate indication in the page when it has started and there is no obvious effect when it has finished. (The content of the dropdown menu changes, but that is not normally visible). Also, it might be problematic to checkpoint a container again that is currently being checkpointed or try to restore a container that hasn't finished checkpointing, depending on how the podman API behaves in these cases. I guess that's why we have the progress indicator here. (The dialog for deleting a container does not have a progress indicator, probably because deletion happens instantly and has an obvious effect.) 

However, the error of a checkpoint action is then shown as a global notification. And retrying is not totally trivial since the user has to ponder all the checkboxes again.

So I'd say we either need to fix the page so that checkpointing can use pattern 2 (put some visible indications somewhere that a container is being checkpointed or has a checkpoint), or use an inline error.

I didn't want to do that here, though.

I guess I simply take out the `this.setState({ inProgress: false });`, okay?

Reply:
> If you take out the inProgress: false from finally(), then pelase move it to .catch() -- then it will DTRT if we ever actually _do_ move the alert into the dialog, and it's the correct thing to do as a best practice anyway.

Excellent idea!

Reply:
Done.